### PR TITLE
Add graceful stop to stop control

### PR DESCRIPTION
**Pull request by pattyshack**
Allow ingestion engine to gracefully stop before crashing.

### CODE REVIEW

File: engine/execution/ingestion/stop/stop_control.go [hunk 1]
```diff
@@ -539,7 +553,16 @@ func (s *StopControl) stopExecution() {
 	log.Warn().Msg("Stopping as finalization reached requested stop")
 
 	if s.stopBoundary.ShouldCrash {
-		// TODO: crash more gracefully or at least in a more explicit way
+		log.Info().Msg(
+			"Attempting graceful stop as finalization reached requested stop")
```
Comment (zhangchiqing) [inline engine/execution/ingestion/stop/stop_control.go:557]: better to calculate the print the timestamp after `s.maxGracefulStopDuration`

**Review by zhangchiqing — APPROVED** [review]

File: engine/execution/ingestion/stop/stop_control.go [hunk 2]
```diff
@@ -19,6 +21,11 @@ import (
 	"github.com/onflow/flow-go/storage"
 )
 
+const (
+	// TODO: figure out an appropriate graceful stop time (is 10 min. enough?)
+	MaxGracefulStopDuration = 10 * time.Minute
+)
```
Comment (fxamacker) [inline engine/execution/ingestion/stop/stop_control.go:27]: Maybe move `MaxGracefulStopDuration` to execution_config.go and unexport it (if it is only used there).

Alternatively, maybe (for consistency with existing code) use the number directly like other default settings in config.

My concern is client code might use this exported variable without realizing it is just the default.

Comment (pattyshack) [reply engine/execution/ingestion/stop/stop_control.go:27]: the code convention is to keep the constant within the module using it.  i'll rename it to DefaultMaxGracefulStopDuration to clarify this.

**Comment by pattyshack** [comment]
bors merge

**Comment by pattyshack** [comment]
bors retry

**Comment by pattyshack** [comment]
bors merge

**Comment by pattyshack** [comment]
bors merge

**Pull request merged**